### ISSUE TITLE
Examples folder: README.md and automated checking?

### ISSUE THREAD

**Issue by fcooper8472** [pints-team/pints#751]
There seem to be a few notebooks in the examples directory that do not appear in the README.md.

Probably unlikely that anyone will open them if they're not in the index, so my plan is to write a script that will check that each example is included in the index.  This would:

* Identify all `*.ipynb` in `pints/examples`
* Check that each filename exists in `README.md`
* Include this in `run_tests.py`
* Run on travis so builds fail if they introduce a non-indexed example

Thoughts?

**Comment by MichaelClerx** [comment]
Love it!

I'd like it if the syntax were something like `./run_tests.py cover-example-readme` and then we could add others such as `./run_tests.py cover-api-docs`. (_Something like that_ better names would be... better). Then we could add a single doc-cover test case to travis (just like we have one for test cover already) that would test all the different kinds of doc-coverage

**Comment by fcooper8472** [comment]
This seems to fit appropriately as a `doctest` in the current `run_tests.py` script.

Currently `python run_tests.py --doctest` was documented as:

```python
"""
Checks if the documentation can be built, runs any doctests (currently not
used).
"""
```

So I have refactored this to `doctest_sphinx()` and `doctest_examples_readme()`, the latter implementing the new functionality.

(Incidentally there are 12 examples not in the `README` at the moment!)

@MichaelClerx any comments on this approach are welcome!

**Comment by fcooper8472** [comment]
Closed by #753
